### PR TITLE
add MySQL to Redshift export job

### PR DESCRIPTION
**Pull request by sureshc**
# Description

Add a script that can be executed manually or on a scheduled basis on `production-daemon` to export data from the production Aurora MySQL cluster to Redshift.  Several iterations of this script have been executed successfully from a development environment to start and monitor the production DMS Replication Tasks once a week for the past 2 months.

https://codedotorg.atlassian.net/browse/INF-281

### Background

This is the first component of several to implement a new Reporting/Analytics export solution that does not rely on MySQL binary logging:

https://docs.google.com/document/d/1WrpOzYGMTUycEfYCk81NVDmvWoFPgimAtYm4QgUPoK0

The additional tasks that will be implemented to complete the analytics export solution are identified in this Epic:
https://codedotorg.atlassian.net/browse/INF-242

### Testing

We don't have a separate test Redshift cluster.  This script has been used to initiate and monitor the weekly production export by executing it on a local development environment once a week for the last 2 months.

## Testing story

High-level logging of export status to Slack #cron-daily chat room works when this process was executed manually on a development environment and allowed to complete after 10 hours:

```
I, [2020-02-21T22:44:56.125091 #90526]  INFO -- : [cron-daily] Beginning export from Aurora MySQL database to Redshift.
I, [2020-02-22T08:22:34.885818 #90526]  INFO -- : [cron-daily] Completed export from Aurora MySQL database to Redshift.
```

Detailed dashboard logs when this process was executed manually on a development environment and completed successfully after 10 hours:

```
Starting DMS Replication Task: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM
Starting DMS Replication Task: arn:aws:dms:us-east-1:475661607190:task:SAFAWX5HX55BWVLOFMFFEBCE6Y
Starting DMS Replication Task: arn:aws:dms:us-east-1:475661607190:task:7CLFRC7A7NMSJSV4UHEVWUOKJ4
Starting DMS Replication Task: arn:aws:dms:us-east-1:475661607190:task:NVDLCOOXKGXLDZKS5ROGNZ3W5A
Starting DMS Replication Task: arn:aws:dms:us-east-1:475661607190:task:WDEGOOQJPLWDDVXLZ3JQSI6ZCI
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: starting / Attempt: 1 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:SAFAWX5HX55BWVLOFMFFEBCE6Y / Status: starting / Attempt: 1 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:NVDLCOOXKGXLDZKS5ROGNZ3W5A / Status: starting / Attempt: 1 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:7CLFRC7A7NMSJSV4UHEVWUOKJ4 / Status: starting / Attempt: 1 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:WDEGOOQJPLWDDVXLZ3JQSI6ZCI / Status: starting / Attempt: 1 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:SAFAWX5HX55BWVLOFMFFEBCE6Y / Status: starting / Attempt: 2 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: starting / Attempt: 2 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:NVDLCOOXKGXLDZKS5ROGNZ3W5A / Status: starting / Attempt: 2 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:WDEGOOQJPLWDDVXLZ3JQSI6ZCI / Status: starting / Attempt: 2 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:7CLFRC7A7NMSJSV4UHEVWUOKJ4 / Status: starting / Attempt: 2 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 3 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:SAFAWX5HX55BWVLOFMFFEBCE6Y / Status: running / Attempt: 3 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:NVDLCOOXKGXLDZKS5ROGNZ3W5A / Status: running / Attempt: 3 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:7CLFRC7A7NMSJSV4UHEVWUOKJ4 / Status: running / Attempt: 3 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:WDEGOOQJPLWDDVXLZ3JQSI6ZCI / Status: running / Attempt: 3 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 4 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:SAFAWX5HX55BWVLOFMFFEBCE6Y / Status: running / Attempt: 4 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:NVDLCOOXKGXLDZKS5ROGNZ3W5A / Status: running / Attempt: 4 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:7CLFRC7A7NMSJSV4UHEVWUOKJ4 / Status: running / Attempt: 4 of 180
DMS Task Completed Successfully: arn:aws:dms:us-east-1:475661607190:task:WDEGOOQJPLWDDVXLZ3JQSI6ZCI
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 5 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:SAFAWX5HX55BWVLOFMFFEBCE6Y / Status: running / Attempt: 5 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:NVDLCOOXKGXLDZKS5ROGNZ3W5A / Status: running / Attempt: 5 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:7CLFRC7A7NMSJSV4UHEVWUOKJ4 / Status: running / Attempt: 5 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 6 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:7CLFRC7A7NMSJSV4UHEVWUOKJ4 / Status: running / Attempt: 6 of 180
DMS Task Completed Successfully: arn:aws:dms:us-east-1:475661607190:task:SAFAWX5HX55BWVLOFMFFEBCE6Y
DMS Task Completed Successfully: arn:aws:dms:us-east-1:475661607190:task:NVDLCOOXKGXLDZKS5ROGNZ3W5A
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 7 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:7CLFRC7A7NMSJSV4UHEVWUOKJ4 / Status: running / Attempt: 7 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 8 of 180
DMS Task Completed Successfully: arn:aws:dms:us-east-1:475661607190:task:7CLFRC7A7NMSJSV4UHEVWUOKJ4
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 9 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 10 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 11 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 12 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 13 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 14 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 15 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 16 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 17 of 180
Replication Task ARN: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM / Status: running / Attempt: 18 of 180
DMS Task Completed Successfully: arn:aws:dms:us-east-1:475661607190:task:GYGKM5GVJKHAUYJF6RONRMXLOM
```

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/cron/export_mysql_database_to_redshift [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+require_relative '../../dashboard/config/environment'
+exit unless rack_env?(:production) && CDO.dashboard_hostname == 'studio.code.org'
+
+require 'aws-sdk-databasemigrationservice'
+require 'cdo/chat_client'
+
+# Export production Aurora MySQL tables to Redshift once a day.
+# 1) Start Data Migration Service (DMS) Replication Tasks, which are configured to drop existing tables in a parallel
+# set of `import_` Redshift schemas and then carry out full load of Aurora MySQL tables into the temporary `import_` schemas.
```
Comment (bencodeorg) [inline bin/cron/export_mysql_database_to_redshift]: Where does (did?) this configuration happen?

Comment (sureshc) [reply bin/cron/export_mysql_database_to_redshift]: Exporting to temporary tables/schemas is not implemented/configured yet.

**Comment by wjordan** [comment]
The logic in this script seems to me to be complicated and long-lasting enough (code we need to maintain over time, not a temporary/one-off script) that it should be written a module and include automated tests covering its behavior.

**Comment by sureshc** [comment]
> The logic in this script seems to me to be complicated and long-lasting enough (code we need to maintain over time, not a temporary/one-off script) that it should be written a module and include automated tests covering its behavior.

DMS logic moved to `cdo/aws/dms` in #32877

**Pull request merged**